src/python/WMComponent/DBS3Buffer/MySQL/CountUndeletedBlocksByWorkflow.py

```diff
@@ -0,0 +1,52 @@
+"""
+_CountUndeletedBlocksByWorkflow_
+
+MySQL implementation of Workflows.CountUndeletedBlocksByWorkflow
+
+Retrieves a list of workflows and the relative undeleted blocks counters,
+"""
+
+
+from WMCore.Database.DBFormatter import DBFormatter
+
+
+class CountUndeletedBlocksByWorkflow(DBFormatter):
+    """
+    Retrieves a list of all workflows and the relative undeleted blocks counters
+
+    The structure returned:
+    [{'count': 6,
+      'deleted': 0,
+      'name': 'PromptReco_Run351572_HcalNZS_Tier0_REPLAY_2022_ID220531142559_v425_220531_1430'},
+     {'count': 8,
+      'deleted': 0,
+      'name': 'PromptReco_Run351572_NoBPTX_Tier0_REPLAY_2022_ID220531142559_v425_220531_1430'},
+      ...]
+    """
+    sql = """
+    SELECT
+        dbsbuffer_workflow.name,
+        dbsbuffer_block.deleted,
+        COUNT(DISTINCT dbsbuffer_block.blockname) as count
+    FROM dbsbuffer_block
+    INNER JOIN dbsbuffer_file ON
+        dbsbuffer_file.block_id = dbsbuffer_block.id
+    INNER JOIN dbsbuffer_workflow ON
+        dbsbuffer_workflow.id = dbsbuffer_file.workflow
+    WHERE dbsbuffer_block.deleted=0
```

Comment:
I don't recall what we discussed when it comes to this `deleted` flag. But we have two options:
1. have this constraint to retrieve only undeleted blocks (deleted=0)
2. or remove it, and actually return a group of deleted and undeleted blocks

If you choose to keep option 1. around, then I do not think we should return the `deleted` column in this query.

Reply:
Hi @amaltaro. We did not accept:
> 2. or remove it, and actually return a group of deleted and undeleted blocks
because it was leading to the following very long query:
```
SQL> SELECT 
  2    undeleted.name, undeleted.count as count_undeleted, deleted.count as count_deleted
  3  FROM ( 
  4  SELECT dbsbuffer_workflow.name, dbsbuffer_block.deleted, COUNT(DISTINCT dbsbuffer_block.blockname) as count
  5  FROM dbsbuffer_block
  6  INNER JOIN dbsbuffer_file ON
  7       dbsbuffer_file.block_id = dbsbuffer_block.id
  8  INNER JOIN dbsbuffer_workflow ON
  9       dbsbuffer_workflow.id = dbsbuffer_file.workflow
 10  INNER JOIN wmbs_workflow ON
 11       dbsbuffer_workflow.name = wmbs_workflow.name
 12  WHERE dbsbuffer_block.deleted = 0
 13  GROUP BY dbsbuffer_workflow.name, dbsbuffer_block.deleted
 14   ) undeleted
 15  LEFT OUTER JOIN ( 
 16  SELECT dbsbuffer_workflow.name, dbsbuffer_block.deleted, COUNT(DISTINCT dbsbuffer_block.blockname) as count
 17  FROM dbsbuffer_block
 18  INNER JOIN dbsbuffer_file ON
 19       dbsbuffer_file.block_id = dbsbuffer_block.id
 20  INNER JOIN dbsbuffer_workflow ON
 21       dbsbuffer_workflow.id = dbsbuffer_file.workflow
 22  INNER JOIN wmbs_workflow ON
 23       dbsbuffer_workflow.name = wmbs_workflow.name
 24  WHERE dbsbuffer_block.deleted = 1
 25  GROUP BY dbsbuffer_workflow.name, dbsbuffer_block.deleted
 26   ) deleted ON
 27    deleted.name = undeleted.name ;
```

We did choose:
> 1. have this constraint to retrieve only undeleted blocks (deleted=0)

And the `Deleted` column is there because we need to `GROUP BY` it.

Reply:
but yes if we avoid the extra group by statement, then Yes we can remove it from the `SELECT` as well

Reply:
ok, tested and removed.